### ISSUE TITLE
Exclude type not preventing to pass "excluded" types

### ISSUE THREAD

**Issue by lazywithclass** [Microsoft/TypeScript#22822]
**TypeScript Version:**  2.8.0-dev.20180322

**Code**

I am experimenting with the `Exclude` type introduced here https://github.com/Microsoft/TypeScript/issues/21847.

What I would like is to create a type that includes all strings except for "", I thought something like this would work:

```TypeScript
type NonEmptyString = Exclude<string, "">;
function test(s: NonEmptyString) {}
test("");
```

so I was expecting an error, but no, it compiles fine. Also `test("a");` compiles fine. I didn't get the reason why, I've asked on gitter and it looks like this code should give a compile error.

**Expected behavior:** I think a compiler error should be shown, but I'm really new with TypeScript, so it might easily be an error on my side

**Actual behavior:** No compiler error, both with `test("")` and `test("a")`

**Comment by ajafff** [comment]
`Exclude` only works with union types. `string` is no union so you would need subtraction types.

Duplicate of #22382

**Comment by lazywithclass** [comment]
I see, thanks.
